docs/standard/microservices-architecture/docker-application-development-process/docker-app-development-workflow.md

```diff
@@ -1,216 +1,339 @@
+
+The initial Dockerfile might look something like this:
+
+```
+ 1  FROM microsoft/dotnet:2.1-aspnetcore-runtime AS base
+ 2  WORKDIR /app
+ 3  EXPOSE 80
+ 4
+ 5  FROM microsoft/dotnet:2.1-sdk AS build
+ 6  WORKDIR /src
+ 7  COPY src/Services/Catalog/Catalog.API/Catalog.API.csproj …
+ 8  COPY src/BuildingBlocks/HealthChecks/src/Microsoft.AspNetCore.HealthChecks … 
+ 9  COPY src/BuildingBlocks/HealthChecks/src/Microsoft.Extensions.HealthChecks …
+10  COPY src/BuildingBlocks/EventBus/IntegrationEventLogEF/ …
+11  COPY src/BuildingBlocks/EventBus/EventBus/EventBus.csproj …
+12  COPY src/BuildingBlocks/EventBus/EventBusRabbitMQ/EventBusRabbitMQ.csproj …
+13  COPY src/BuildingBlocks/EventBus/EventBusServiceBus/EventBusServiceBus.csproj …
+14  COPY src/BuildingBlocks/WebHostCustomization/WebHost.Customization …
+15  COPY src/BuildingBlocks/HealthChecks/src/Microsoft.Extensions …
+16  COPY src/BuildingBlocks/HealthChecks/src/Microsoft.Extensions …
+17  RUN dotnet restore src/Services/Catalog/Catalog.API/Catalog.API.csproj
+18  COPY . .
+19  WORKDIR /src/src/Services/Catalog/Catalog.API
+20  RUN dotnet build Catalog.API.csproj -c Release -0 /app
+21
+22  FROM build AS publish
+23  RUN dotnet publish Catalog.API.csproj -c Release -0 /app
+24
+25  FROM base AS final
+26  WORKDIR /app
+27  COPY --from=publish /app
+28  ENTRYPOINT ["dotnet", "Catalog.API.dll"]
+```
+And these are the details, line by line:
+
+1.  Begin a stage with a "small" runtime-only base image, call it **base** for reference.
+2.  Create **/app** directory in the image.
+3.  Expose port **80**.
+<!-- skip -->
+5.  Begin a new stage with "large" image for building/publishing, call it **build** for reference.
+6.  Create directory **/src** in the image.
+7.  Up to line 16, copy referenced projects **.csproj** files, to be able to restore pakages later.
+<!-- skip -->
+17. Restore packages for the **Catalog.API** project and the referenced projects.
+18. Copy **all directory tree for the solution** (except the files/directories included in the **.dockerignore** file) from to the **/src** directory in the image.
+19. Change current folder to **Catalog.API** project.
+20. Build project (and other project dependencies) and output to **/app** directory in the image.
+<!-- skip -->
+22. Begin a new stage continuing from build, call it **publish** for reference.
+23. Publish project (and dependencies) and output to **/app** directory in the image.
+<!-- skip -->
+25. Begin a new stage continuing from **base** and call it **final**
+26. Change current directory to **/app**
+27. Copy the **/app** directory from stage **publish** to the current directory
+28. Define the command to run when the container is started.
+
+Now we'll explore some optimizations to improve the whole process performance that, in the case of eShopOnContainers, means about 22 minutes or more to build the complete solution in Linux containers.
```

Comment:
remove we from these sentences here and in the next paragraph. Maybe let's explore

Reply:
done.